exports/dataset.cpp

```diff
@@ -926,12 +979,20 @@ PYBIND11_MODULE(dataset, m) {
             return self(key.first, key.second);
           },
           py::keep_alive<0, 1>())
-      .def("copy", [](const DatasetSlice &self) { return Dataset(self); })
+      .def("copy", [](const DatasetSlice &self) { return Dataset(self); },
+           "Make a copy of a DatasetSlice.")
       .def("__copy__", [](DatasetSlice &self) { return Dataset(self); })
       .def("__deepcopy__",
            [](DatasetSlice &self, py::dict) { return Dataset(self); })
       .def_property_readonly(
-          "subset", [](DatasetSlice &self) { return SubsetHelper(self); })
+          "subset", [](DatasetSlice &self) { return SubsetHelper(self); },
+          "Used to extract a read-only subset of the DatasetSlice in two "
+          "different ways: \n - one can use just a string, e.g. "
+          "d.subset['sample'] to extract the sample and its variance, and all "
+          "the relevant coordinates.\n - one can also use a tag/string "
+          "combination for a more restrictive extraction, e.g. "
+          "d.subset[Data.Value, 'sample'] to only get the Value and associated "
+          "coordinates.")
```

Comment:
Generally good. Is it worth stating what happens to attribute variables when the named subset option is used?

Reply:
Can you tell me what happens to them So i can add it, thanks.

Reply:
Looking at the code, I think they are not kept. Theres actually a comment about that here https://github.com/mantidproject/dataset/blob/master/src/dataset.h#L332